adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/jaas/AbstractKeycloakLoginModule.java

```diff
@@ -84,14 +88,30 @@ public void initialize(Subject subject, CallbackHandler callbackHandler, Map<Str
 
     protected KeycloakDeployment resolveDeployment(String keycloakConfigFile) {
         try {
-            InputStream is = FindFile.findFile(keycloakConfigFile);
+            InputStream is = null;
```

Comment:
Why are you not just checking if keycloakConfigFile starts with http:// or https:// and use URL in that case, or FindFile.findFile if not?

Reply:
@stianst because the current usage is to only support resources profile:resourcename which will exist in JBoss Fuse Fabric profile resource. In Karaf environment the resource URL profile:resourcename is resolved as it searches it in the fabric profile for the resource. As per @mstruk we should avoid allowing any other protocols like http or https as it may search for the file not existing locally. So hence I have added the check only for the profile.

Reply:
Ok, that makes sense. I would still suggest you do a string check first to see if keycloakConfigFile starts with 'profile:' rather than rely on exception being thrown.

Reply:
Ok @stianst I will implement the changes accordingly. Thanks for the help.

Reply:
@stianst Changes implemented.

Reply:
My original suggestion with exception was to completely avoid checking for 'protocol:' in order to allow potential integration with other frameworks OOTB. In that sense if we want to prevent going to the internet we could just check for http, https, and ftp, and let all else through. Although actually that solution is not future proof - if some additional network protocol becomes available through URL class, we would have to keep an eye on it and add it to exclude list. So yeah, might be better to go with include list instead.

Reply:
@stianst Could you please review the changes done.